### ISSUE TITLE
Module for Nanoplot

### ISSUE THREAD

**Issue by FranBonath** [ewels/MultiQC#1074]
**Tool Details**
 - Name of tool: Nanoplot
 - Tool homepage: https://github.com/wdecoster/NanoPlot
 - Tool description: Plotting tool for long read sequencing data and alignments
 - Data suitable for MultiQC plot(s): plots and histograms
 - Most interesting data for General Stats table: read length N50, mean read length and mean read quality, number of reads

**Tool output:**
 - Log filename pattern: `NanoStats.txt`
 - File contents: [NanoStats.txt](https://github.com/ewels/MultiQC/files/3922479/NanoStats.txt)

**Comment by ewels** [comment]
Nice! And you are writing this module, right?

**Comment by FranBonath** [comment]
Yes, with time...

**Comment by hoelzer** [comment]
Hi, thanks for the effort to integrate Nanoplot into MultiQC! It would be great to have this, for example, also to summarize multiple Nanoplot runs for different barcodes.

**Comment by RxLoutre** [comment]
Hello everyone ! I might be late but I would still be very interested by Nanoplot being added in multi_qc :) . Is there any progress ? Thank you for your work ! If help is needed, maybe I could give a hand :)

**Comment by ewels** [comment]
Hi all,

NanoStat is now supported thanks to the new module written by @kpalin, just merged in #1344 🎉 It will be available in the upcoming v1.11 MultiQC release.

@FranBonath - in your original issue you named NanoPlot, but the module is NanoStat. However, the file you attached looks identical, so I'm assuming that running NanoPlot also runs NanoStat internally and generates the same file. Hopefully this is clear enough.

Note that some of the plots in NanoPlot could be nice to reproduce in MultiQC, for example line graphs showing read length histograms per sample especially (also read qualities, etc). I'm not sure if NanoPlot dumps any of these into a text file that we can add support for, if it does then that would be great 👍🏻  If anyone reading this has such a file, please create a new issue for it and we can look into adding support.

Thanks all,

Phil

**Comment by MarieLataretu** [comment]
Hi all,

I'm very happy to see a NanoStats module - thanks to all the involved people!

---

As we integrated NanoPlot into our pipeline, I tried to include the `NanoStats.txt` file (produced by NanoPlot) in the MultiQC report.
Unfortunately, there is a slight difference in the general summary report, so that the file is not recognized as NanoStats file: [NanoStats_fromNanoPlot.txt](https://github.com/ewels/MultiQC/files/6856727/A2_read_quality.txt)

---

> I'm not sure if NanoPlot dumps any of these into a text file that we can add support for, if it does then that would be great

The NanoPlot readme offers this:
```
-f, --format          Specify the output format of the plots.
                        Default = png, other options: jpg,jpeg,webp,svg,pdf,eps,json. Saving the figure as a json file allows for further customisation and can be plotted locally with plotly (https://plotly.com/python-api-reference/generated/plotly.io.read_json.html).
```
